onyo/cli/tests/test_new.py

```diff
@@ -284,9 +284,31 @@ def test_keys_flag(repo: OnyoRepo, directory: str) -> None:
     assert repo.git.is_clean_worktree()
 
 
+def test_error_keys_flag_mismatch_count(repo: OnyoRepo) -> None:
+    r"""
+    Test that `onyo new --keys KEY=VALUE` requires that repeated keys are given
+    the same number of times.
+    """
+    key_values = asset_spec + ['serial=1', 'mode=0', 'mode=1', 'mode=2']
```

Comment:
I know it is a nitpick, but I find this a bit un-readable, since it is not obvious that "asset_spec" contains also a serial, and that with the added second serial and the three modes the error is triggered.

My first thought was "well, here we have something 1 time (serial) and another thing 3 times (mode) -- this should work just fine". So I would prefer e.g. either having instead of "mode" 2 times "make", or alternatively, instead of "serial" two times something else not in asset_spec.

Either way it is just a small change that would make it easier visible why the error is triggered, when not part is from the `asset_spec` variable and part stated explicitely :)

Reply:
I agree that it's non-obvious. I was trying to follow the conventions in this file, which I find in general to be non-obvious.

I have been told that there's a desire for the tests to be a similar as possible and not re-define the same things over and over. So I followed that convention.

>  or alternatively, instead of "serial" two times something else not in asset_spec.

Serial needs to be multiple times, because the multiple assets need to be unique.